### ISSUE TITLE
Flaky crashes with "scavenger.cc: 1154: error: expected: external_size_ >= 0" on Windows

### ISSUE THREAD

**Issue by sjindel-google** [dart-lang/sdk#40751]
This has shown up in two tests:

```
/========================================================================================\
| standalone_2/io/file_system_watcher_test/0 failed again (Crash -> Fail, expected Pass) |
\========================================================================================/

--- Command "vm" (took 07.000312s):
set DART_CONFIGURATION=DebugX64 & set DART_SUPPRESS_WER=1 & set DART_CRASHPAD_HANDLER=C:\b\s\w\ir\out\DebugX64\crashpad_handler.exe & out\DebugX64\dart.exe --enable-isolate-groups --ignore-unrecognized-flags --packages=C:\b\s\w\ir\.packages C:\b\s\w\ir\tests\standalone_2\io\file_system_watcher_test.dart

exit code:
-36862

stdout:
unittest-suite-wait-for-done

stderr:
c:\b\s\w\ir\cache\builder\sdk\runtime\vm\heap\scavenger.cc: 1154: error: expected: external_size_ >= 0
c:\b\s\w\ir\cache\builder\sdk\runtime\vm\heap\scavenger.cc: 1154: error: expected: external_size_ >= 0
version=2.8.0-edge.d41e41990401af0019b930776b78f3c8f281b628 (Sat Feb 22 01:20:21 2020 +0000) on "windows_x64"
thread=11260, isolate=main(0000000002C23CF0)
Aborting re-entrant request for stack trace.
isolate_instructions=140055070, vm_instructions=140055080
  pc 0x000000014061615c fp 0x000000001089f020 Dart_HandleServiceMessages
-- End of DumpStackTrace
[exit     : sp(0) fp(0x1089f208) pc(0)]
[dart     : sp(0x1089f218) fp(0x1089f248) pc(0x3a7bb53) dart:io__RandomAccessFileOpsImpl@15069316_close ]
[dart     : sp(0x1089f258) fp(0x1089f290) pc(0x3a7b8a6) dart:io__RandomAccessFile@15069316_closeSync ]
[dart     : sp(0x1089f2a0) fp(0x1089f318) pc(0x3a73bf8) dart:io__File@15069316_writeAsBytesSync ]
[dart     : sp(0x1089f328) fp(0x1089f3a0) pc(0x3a73174) dart:io__File@15069316_writeAsStringSync ]
[dart     : sp(0x1089f3b0) fp(0x1089f438) pc(0x10778d32) file:///C:/b/s/w/ir/tests/standalone_2/io/file_system_watcher_test.dart_::_testWatchOverflow__async_op_<anonymous closure>__async_op ]
[dart     : sp(0x1089f448) fp(0x1089f470) pc(0x3a3f087) dart:core__Closure@0150898_call ]
[dart     : sp(0x1089f480) fp(0x1089f4a8) pc(0xe7093b6) dart:async__AsyncAwaitCompleter@4048458_start ]
[dart     : sp(0x1089f4b8) fp(0x1089f510) pc(0x10778398) file:///C:/b/s/w/ir/tests/standalone_2/io/file_system_watcher_test.dart_::_testWatchOverflow__async_op_<anonymous closure> ]
[dart     : sp(0x1089f520) fp(0x1089f598) pc(0xe713a0e) dart:async__RootZone@4048458_runUnaryGuarded ]
[dart     : sp(0x1089f5a8) fp(0x1089f5f0) pc(0xe71d60b) dart:async__BufferingStreamSubscription@4048458__sendData@4048458 ]
[dart     : sp(0x1089f600) fp(0x1089f630) pc(0xe71d15b) dart:async__BufferingStreamSubscription@4048458__add@4048458 ]
[dart     : sp(0x1089f640) fp(0x1089f670) pc(0xe71ca9c) dart:async__SyncStreamController@4048458__sendData@4048458 ]
[dart     : sp(0x1089f680) fp(0x1089f6b0) pc(0xe71c709) dart:async__StreamController@4048458__add@4048458 ]
[dart     : sp(0x1089f6c0) fp(0x1089f6f0) pc(0xe71c21c) dart:async__StreamController@4048458_add ]
[dart     : sp(0x1089f700) fp(0x1089f740) pc(0x10745797) dart:async__StreamController@4048458_add_add ]
[dart     : sp(0x1089f750) fp(0x1089f780) pc(0x3a3dffa) dart:core__Closure@0150898_call ]
[dart     : sp(0x1089f790) fp(0x1089f7b8) pc(0x3a3defe) dart:isolate__RawReceivePortImpl@1026248__handleMessage@1026248 ]
[entry    : sp(0x1089f7c8) fp(0x1089f8d8) pc(0x2501136)]
c:\b\s\w\ir\cache\builder\sdk\runtime\vm\thread.h: 773: error: expected: no_safepoint_scope_depth() == 0
Aborting re-entrant request for stack trace.

--- Re-run this test:
python tools/test.py -n dartk-win-debug-x64 standalone_2/io/file_system_watcher_test/0
```
https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket.appspot.com/8887794976384732992/+/steps/test_results/0/logs/ignored_flaky_test_failure_logs/0

```
/=====================================================================================\
| standalone_2/io/socket_many_connections_test/0 broke (Pass -> Crash, expected Pass) |
\=====================================================================================/

--- Command "vm" (took 17.000843s):
set DART_CONFIGURATION=DebugIA32 & set DART_SUPPRESS_WER=1 & set DART_CRASHPAD_HANDLER=C:\b\s\w\ir\out\DebugIA32\crashpad_handler.exe & out\DebugIA32\dart.exe --enable-isolate-groups --ignore-unrecognized-flags --packages=C:\b\s\w\ir\.packages C:\b\s\w\ir\tests\standalone_2\io\socket_many_connections_test.dart

exit code:
1073741845

stdout:
unittest-suite-wait-for-done
unittest-suite-success

stderr:
c:\b\s\w\ir\cache\builder\sdk\runtime\vm\heap\scavenger.cc: 1154: error: expected: external_size_ >= 0
version=2.8.0-edge.55f4b1294eb6ed57e57a209204efdff125c632f4 (Sat Feb 22 02:22:21 2020 +0000) on "windows_ia32"
thread=644, isolate=startTestServer(04AA2B18)
isolate_instructions=454d60, vm_instructions=454d70
  pc 0x008f409e fp 0x06c1fb3c Unknown symbol
  pc 0x004a06c9 fp 0x06c1fb48 Unknown symbol
  pc 0x00c3718f fp 0x06c1fb64 Unknown symbol
  pc 0x00c1e0b8 fp 0x06c1fb78 Unknown symbol
  pc 0x00c3a7d2 fp 0x06c1fb98 Unknown symbol
  pc 0x0079fde8 fp 0x06c1fbb8 Unknown symbol
  pc 0x00c3984b fp 0x06c1fd90 Unknown symbol
  pc 0x00c36cd2 fp 0x06c1fdb0 Unknown symbol
  pc 0x00c1d001 fp 0x06c1fdf4 Unknown symbol
  pc 0x0079db94 fp 0x06c1fe60 Unknown symbol
  pc 0x007edad6 fp 0x06c1fe98 Unknown symbol
  pc 0x007ed622 fp 0x06c1fea8 Unknown symbol
  pc 0x009caf97 fp 0x06c1fef4 Unknown symbol
  pc 0x009cb2d4 fp 0x06c1ff14 Unknown symbol
  pc 0x008ea936 fp 0x06c1ff38 Unknown symbol
  pc 0x00fb7252 fp 0x06c1ff80 Unknown symbol
  pc 0x74d88744 fp 0x06c1ff94 Unknown symbol
  pc 0x777a582d fp 0x06c1ffdc Unknown symbol
  pc 0x777a57fd fp 0x06c1ffec Unknown symbol
-- End of DumpStackTrace

--- Re-run this test:
python tools/test.py -n dartk-win-debug-ia32 standalone_2/io/socket_many_connections_test/0
```
https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket.appspot.com/8887788938818891104/+/steps/test_results/0/logs/ignored_flaky_test_failure_logs/0

/cc @mkustermann

**Comment by sjindel-google** [comment]
This seems related to https://dart-review.googlesource.com/c/sdk/+/126646 as it changes `FinalizablePersistentHandle::SetExternalSize` and `FinalizablePersistentHandle::EnsureFreeExternal`.

**Comment by mkustermann** [comment]
For other readers of this issue: This test is specifically opting into the experimental `--enable-isolate-groups` flag.

**Comment by mkustermann** [comment]
This was very likely fixed by 056bbd88f93323a50153ad1e25fc2b5eb1c9f8db
